cuda_core/cuda/core/experimental/_module.py

```diff
@@ -46,26 +73,29 @@ def __init__(self, module, code_type, jit_options=None, *,
                  symbol_mapping=None):
         if code_type not in self._supported_code_type:
             raise ValueError
+        _lazy_init()
         self._handle = None
 
-        driver_ver = handle_return(cuda.cuDriverGetVersion())
-        self._loader = _backend["new"] if driver_ver >= 12000 else _backend["old"]
+        backend = "new" if (_py_major_ver >= 12 and _driver_ver >= 12000) else "old"
+        self._loader = _backend[backend]
 
         if isinstance(module, str):
-            if driver_ver < 12000 and jit_options is not None:
+            # TODO: this option is only taken by the new library APIs, but we have
+            # a bug that we can't easily support it just yet (NVIDIA/cuda-python#73).
+            if jit_options is not None:
                 raise ValueError
             module = module.encode()
             self._handle = handle_return(self._loader["file"](module))
         else:
             assert isinstance(module, bytes)
             if jit_options is None:
                 jit_options = {}
-            if driver_ver >= 12000:
+            if backend == "new":
                 args = (module, list(jit_options.keys()), list(jit_options.values()), len(jit_options),
                         # TODO: support library options
                         [], [], 0)
-            else:
-                args = (module, len(jit_options), jit_options.keys(), jit_options.values())
+            else:  # "old" backend
+                args = (module, len(jit_options), list(jit_options.keys()), list(jit_options.values()))
```

Comment:
btw @vzhurba01 this is in the same class of "bugs" as described in https://github.com/NVIDIA/cuda-python/issues/72, only caught until CUDA 11 support is added back in this PR. I think the codegen might be too restrictive in only allowing lists as the only valid sequence container...

Reply:
(mark as resolved to make web UI check happy)

Reply:
> https://github.com/NVIDIA/cuda-python/blob/11.8.x/cuda_bindings/cuda/bindings/driver.pyx.in#L15883
> https://github.com/NVIDIA/cuda-python/blob/main/cuda_bindings/cuda/bindings/driver.pyx.in#L20655

I don't think I understand. This should have already been fixed for both 11.x and 12.x. Perhaps what you hit was us missing the 11.8.4 patch release that would have backported a lot of the fixes such as this one?

Reply:
I see, so it's already backported. It could be possible that I am still using public builds of the bindings instead of building from source. But it also means we need to keep WARs like these for a little longer, until either older bindings are less used, or cuda-core can pin tightly at cuda-bindings (#105).